x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceIndexSettingsProvider.java

```diff
@@ -67,28 +71,67 @@ public Settings getAdditionalIndexSettings(
         IndexMode templateIndexMode,
         Metadata metadata,
         Instant resolvedAt,
-        Settings indexTemplateAndCreateRequestSettings,
+        Settings settings,
         List<CompressedXContent> combinedTemplateMappings
     ) {
-        var logsdbSettings = logsdbIndexModeSettingsProvider.getLogsdbModeSetting(dataStreamName, indexTemplateAndCreateRequestSettings);
+        var logsdbSettings = logsdbIndexModeSettingsProvider.getLogsdbModeSetting(dataStreamName, settings);
         if (logsdbSettings != Settings.EMPTY) {
-            indexTemplateAndCreateRequestSettings = Settings.builder()
-                .put(logsdbSettings)
-                .put(indexTemplateAndCreateRequestSettings)
-                .build();
+            settings = Settings.builder().put(logsdbSettings).put(settings).build();
         }
 
         // This index name is used when validating component and index templates, we should skip this check in that case.
         // (See MetadataIndexTemplateService#validateIndexTemplateV2(...) method)
         boolean isTemplateValidation = "validate-index-name".equals(indexName);
-        if (newIndexHasSyntheticSourceUsage(indexName, templateIndexMode, indexTemplateAndCreateRequestSettings, combinedTemplateMappings)
-            && syntheticSourceLicenseService.fallbackToStoredSource(isTemplateValidation)) {
-            LOGGER.debug("creation of index [{}] with synthetic source without it being allowed", indexName);
-            return Settings.builder()
-                .put(SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.getKey(), SourceFieldMapper.Mode.STORED.toString())
-                .build();
-        }
+        if (syntheticSourceLicenseService.fallbackToStoredSource(isTemplateValidation)) {
+            var builder = Settings.builder();
+            if (newIndexHasSyntheticSourceUsage(indexName, templateIndexMode, settings, combinedTemplateMappings)) {
+                LOGGER.debug("creation of index [{}] with synthetic source without it being allowed", indexName);
+                builder.put(SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.getKey(), SourceFieldMapper.Mode.STORED.toString());
+            }
+            if (settings.getAsBoolean(IndexSettings.LOGSDB_ROUTE_ON_SORT_FIELDS.getKey(), false)) {
+                builder.put(IndexSettings.LOGSDB_ROUTE_ON_SORT_FIELDS.getKey(), false);
+            }
+            return builder.keys().isEmpty() ? Settings.EMPTY : builder.build();
+        } else if (templateIndexMode == IndexMode.LOGSDB
```

Comment:
So the idea here is to only allow route on sort fields if synthetic source usage is allowed?
I think we shouldn't convolute synthetic source and routing on sort fields. I think we should handle this separately before handling synthetic source (line 82) and then create a separate licensed feature. For the latter we can create a class like `SyntheticSourceLicenseService` or add a method to `SyntheticSourceLicenseService` for handling licensing for routing on sort fields.

Reply:
Hm this keeps getting more and more complicated.. Let's check with Tyler if we want to make this part of licensing and how.

Reply:
I left the fallback logic per license as a follow-up, to keep this one simpler. You are right, we need a separate licensed feature here.